### PR TITLE
64 api

### LINKED ISSUE #64: 64 Create new API
**Please describe**
So we have our current API (not in develop yet), but I am already getting sick of it. There are some functions like: Select, Create and Update. The problem I can't figure out how to return the result from outside the function.
Currently we don't have a good security, well we don't have any.

**Describe the solution you would like**
I am going to remove the functions and just write it out. This might not be D.R.Y. but you can do something with the returned data. Also I am going to be writing object for each route like this:
```
const user = {
  id: {
    type: 'string',
    required: true
  }
}
```
The reason I will do this is so it is clear what the DB looks like and we can also check for types and give more clear error messages to the font-end.
The api's will use passport.js and express-sessions. When or if a user is not logged you will get an error messages (403, User is not logged in)  We do have to decide which api's can and can't be called by users that are not logged in. I will be using the local strategy.

**Describe alternatives you have considered**
For the api's I have tried making the functions, in the functions pushing it in an array. There are a lot of things I tried most of it I forgot.
For security I thought about JWT (JSONWebToken) but I am not using JWT because passport.js is using JWT but more secure.

**Additional context**
If you want to take a look at passport yourself:
http://www.passportjs.org
http://www.passportjs.org/packages/passport-local/

**Definition Of Done**
It will be declared done when:
- [ ] The api's: User, Posts and Role have been made.
- [ ] Passport.js has been implented in the previous API's

### PR DESCRIPTION
**Pull request by vQFire**
closes #64 

 **What is it supposed to do**
You should be able to fetch data from the API

 **How can we test your branch**
Install insomnia import the insomnia.json file (`system/Insomnia`) Then all the router will be available.
Make sure you have XAMMP running with the database (`system/database`)
Then run `npm ci`
Then run `npm run serve`

 **What broweser have been checked**
No browser check needed, because it is backend

### CODE REVIEW

File: system/controllers/post.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  },
+  title: {
+    type: 'string',
+    required: false
+  },
+  content: {
+    type: 'string',
+    required: false
+  },
```
Comment (MiguelFieira) [inline system/controllers/post.js:23]: Just a Question
Is this the template for a post? If so where do we get the post banner from?

File: system/controllers/post.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    return res.status(412).send( { 'msg': 'ID has not been defined' } )
+  }
+
+  Post.archive(body.id, () => {
+    res.send( { 'msg': `User with ID: ${body.id} has been archived` } )
+  })
+})
+
+
+module.exports = router
```
Comment (MiguelFieira) [inline system/controllers/post.js]: Endline

**Review by MiguelFieira — COMMENTED** [review]
A question and 1 linting that should be fixed ;)

File: system/controllers/post.js (hunk 1 above)
Comment (vQFire) [reply system/controllers/post.js:23]: This is a template to make sure all the correct info is given when creating or updating a post. I have made it like this so we can expand it really easy!

**Comment by AjobK** [comment]
You would think about the future of the Seaqull back-end. Please notify me. Until that time this PR is closed

**Pull request closed without merging**